trackpy/linking/subnet.py

```diff
@@ -10,6 +10,7 @@
 
 try:
     from sklearn.neighbors import BallTree
+    from sklearn.metrics import DistanceMetric
```

Comment:
This line appears to raise an `ImportError` in many of our automated tests, which in turn breaks `BallTree` even though `sklearn` is installed and detected. Here is an example: https://github.com/soft-matter/trackpy/runs/5127629137?check_suite_focus=true#step:6:2336 . Is `DistanceMetric` absent from earlier sklearn versions?

Reply:
Indeed, it appears to be the difference between e.g. scikit-learn 1.0.2 and scikit-learn 0.24.2 . Since DistanceMetric was only released in late 2021, this code should check for its presence separately from BallTree.

Reply:
Thanks for interpreting the CI output, I was seeing a failure but was having a hard time deducing the cause without further investigation. Looking through the API versions, it looks like they recently moved `DistanceMetric` over to `sklearn.metrics`. Before, it was in `sklearn.neighbors`. I'll update the PR to resolve this- first try importing `DistanceMetric` from `sklearn.metrics` and if an `ImportError` is raised, import it from `sklearn.neighbors`. It looks like `sklearn.neighbors.DistanceMetric` should work as far back as at least scikit-learn 0.16 (March 2015); should I still have it separately check for presence, or is it safe to assume that users will have a version newer than this? Happy to do either, but of course assuming that it is there is easier.